test/lib/ansible_test/_data/sanity/pylint/plugins/deprecated.py

```diff
@@ -125,17 +154,71 @@ class AnsibleDeprecatedChecker(BaseChecker):
 
     def __init__(self, *args, **kwargs):
         self.collection_version = None
-        self.is_collection = False
-        self.version_constructor = LooseVersion
+        self.collection_name = None
         super(AnsibleDeprecatedChecker, self).__init__(*args, **kwargs)
 
     def set_option(self, optname, value, action=None, optdict=None):
         super(AnsibleDeprecatedChecker, self).set_option(optname, value, action, optdict)
         if optname == 'collection-version' and value is not None:
-            self.version_constructor = SemanticVersion
-            self.collection_version = self.version_constructor(self.config.collection_version)
-        if optname == 'is-collection':
-            self.is_collection = self.config.is_collection
+            self.collection_version = SemanticVersion(self.config.collection_version)
+        if optname == 'collection-name' and value is not None:
+            self.collection_name = self.config.collection_name
+
+    def _check_date(self, node, date):
+        if not isinstance(date, str):
+            self.add_message('invalid-tagged-date', node=node, args=(date,))
+            return
+
+        matcher = TAGGED_VERSION_RE.match(date)
+        if not matcher:
+            self.add_message('invalid-tagged-date', node=node, args=(date,))
+            return
+
+        collection = matcher.group(1)
+        date_str = matcher.group(2)
+
+        if collection != (self.collection_name or 'ansible.builtin'):
+            self.add_message('wrong-collection-deprecated-date-tag', node=node, args=(date,))
+
+        try:
+            if parse_isodate(date_str) < datetime.date.today():
+                self.add_message('ansible-deprecated-date', node=node, args=(date,))
+        except ValueError:
+            self.add_message('ansible-invalid-deprecated-date', node=node, args=(date,))
+
+    def _check_version(self, node, version):
+        if not isinstance(version, str):
+            self.add_message('invalid-tagged-version', node=node, args=(version,))
+            return
+
+        matcher = TAGGED_VERSION_RE.match(version)
+        if not matcher:
+            self.add_message('invalid-tagged-version', node=node, args=(version,))
+            return
+
+        collection = matcher.group(1)
+        version_no = matcher.group(2)
+
+        if collection != (self.collection_name or 'ansible.builtin'):
+            self.add_message('wrong-collection-deprecated-version-tag', node=node, args=(version,))
+
+        if collection == 'ansible.builtin':
```

Comment:
Shouldn't this be checking `self.collection_name` instead?

You could use the following (but the collection block would need to come first then):

```suggestion
        if self.collection_name:
```

Reply:
No, since `ansible.builtin:2.14` should not trigger an `ansible-invalid-deprecated-version` or `ansible-deprecated-version` error, but only an `wrong-collection-deprecated-version-tag` error.

(If someone uses `ansible.builtin:1.0.0`, they'll get more errors.)